components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -56,6 +60,27 @@ CurlDownloadHandler::CurlDownloadHandler(
         m_http_headers.append("Cache-Control: no-cache");
         m_http_headers.append("Pragma: no-cache");
     }
+    std::set<std::string> const kReservedHeaders = {"range", "cache-control", "pragma"};
+    // RFC 7230 token pattern: one or more tchars
+    std::regex const header_name_pattern("^[!#$%&'*+.^_`|~0-9a-zA-Z-]+$");
+    // Must consist of printable ASCII characters (values between 0x20 and 0x7E)
+    std::regex const header_value_pattern("^[\\x20-\\x7E]*$");
+    for (auto const& [key, value] : custom_headers) {
+        // Convert to lowercase for case-insensitive comparison
+        std::string lower_key = key;
+        std::transform(lower_key.begin(), lower_key.end(), lower_key.begin(), [](unsigned char c) {
+            return std::tolower(c);
+        });
+
+        if (kReservedHeaders.end() == kReservedHeaders.find(lower_key)) {
+            // Filter out illegal header names and header values by regex
+            if (std::regex_match(key, header_name_pattern)
+                && std::regex_match(value, header_value_pattern))
+            {
+                m_http_headers.append(key + ": " + value);
+            }
+        }
```

Comment:
I think if we found a reserved key we should throw an exception instead of swallowing the warning. What do you think? @kirkrodrigues

Reply:
Yep.

Reply:
What kind of exception shall I throw? (or just std::runtime_error?)

Reply:
Please check CurlOperationFailed